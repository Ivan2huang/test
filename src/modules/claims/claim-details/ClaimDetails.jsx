import React, { useEffect } from 'react';
import { injectIntl } from 'react-intl';
import { withRouter } from 'next/router';
import * as PropTypes from 'prop-types';

import { Box, Card, CardContent, withStyles, Button } from '@material-ui/core';

import GridItem from '../../../uiComponents/GridItem';
import Grid from '../../../uiComponents/Grid';
import Typography from '../../../uiComponents/Typography';
import NotificationBox from '../../../uiComponents/NotificationBox';
import { formatMessage } from '../../../helpers/helpers';
import Details from './Details';
import { CLAIM_STATUS, CLAIM_TYPES } from '../constant';
import { NOTIFICATION_TYPES } from '../../../constants/types';
import {
  claimDetailsWithIntl,
  claimSubmissionDetails,
  getBackgroundImage,
  patientDetails,
} from './helper';

import { AlertIcon } from '../../../icons';

const claimStatusDetails = {
  [CLAIM_STATUS.PENDING]: {
    headerId: 'claims.claimDetails.status.processing',
    headerDefaultMessage: 'Processing',
    displayRemark: false,
  },
  [CLAIM_STATUS.REQUEST_FOR_INFORMATION]: {
    headerId: 'claims.claimDetails.status.processing',
    headerDefaultMessage: 'Processing',
    displayRemark: true,
    remarkId: 'claims.claimDetails.status.processing.remark',
    remarkDefaultMessage:
      'More information is required. We will contact you soon.',
    remarkIcon: <AlertIcon size={24} />,
    remarkType: NOTIFICATION_TYPES.ERROR,
  },
  [CLAIM_STATUS.APPROVED]: {
    headerId: 'claims.claimDetails.status.approved',
    headerDefaultMessage: 'Approved',
    displayRemark: false,
  },
  [CLAIM_STATUS.REJECTED]: {
    headerId: 'claims.claimDetails.status.rejected',
    headerDefaultMessage: 'Rejected',
    displayRemark: false,
  },
};

const Styles = theme => ({
  cardRoot: {
    [theme.breakpoints.down('sm')]: {
      background: 'none',
      boxShadow: 'none',
    },
  },
  cardContent: {
    '&:last-child': {
      paddingBottom: theme.spacingX(10),
    },
    [theme.breakpoints.up('md')]: {
      padding: `${theme.spacingX(10)} ${theme.spacingX(9)}`,
      '&:last-child': {
        paddingBottom: theme.spacingX(18),
      },
    },
  },
  remark: {
    backgroundColor: `${theme.grey1}`,
    padding: `${theme.spacingX(3)} ${theme.spacingX(5)}`,
    marginTop: `${theme.spacingX(5)}`,
    display: 'inline-block',
    fontWeight: 500,
  },
  moreInfo: {
    marginTop: theme.spacingX(10),
  },
});

const ClaimDetails = ({
  intl,
  classes,
  router,
  claimDetails,
  membersMap,
  getNewClaimDetails,
  getMemberProfile,
  claims,
}) => {
  const { statusCode, isCashlessClaim, claimItemCategoryCode } = claimDetails;
  const claimType =
    claimItemCategoryCode === CLAIM_TYPES.OUTPATIENT
      ? formatMessage(
          intl,
          'claims.claimDetails.subHeader.outpatient',
          'Outpatient Claim',
        )
      : formatMessage(
          intl,
          'claims.claimDetails.subHeader.wellness',
          'Wellness Claim',
        );

  useEffect(() => {
    getNewClaimDetails(router.query.id, claims);
    getMemberProfile();
  }, []);

  const backgroundImage = getBackgroundImage(statusCode);
  const renderHeader = () => (
    <Box mt={{ xs: 2, md: 0 }} textAlign={{ xs: 'center', md: 'left' }}>
      <Typography type="style-1">
        {formatMessage(
          intl,
          claimStatusDetails[statusCode].headerId,
          claimStatusDetails[statusCode].headerDefaultMessage,
        )}
      </Typography>
      <Typography type="style-6">{claimType}</Typography>
      {!isCashlessClaim && claimDetails.remark && (
        <Typography classes={{ root: classes.remark }}>
          {claimDetails.remark}
        </Typography>
      )}
    </Box>
  );

  const renderRemark = () => {
    return (
      claimStatusDetails[statusCode].displayRemark && (
        <NotificationBox
          className={classes.moreInfo}
          type={claimStatusDetails[statusCode].remarkType}
        >
          <Box pr={2}>{claimStatusDetails[statusCode].remarkIcon}</Box>
          <Typography type="style-5">
            {formatMessage(
              intl,
              claimStatusDetails[statusCode].remarkId,
              claimStatusDetails[statusCode].remarkDefaultMessage,
            )}
          </Typography>
        </NotificationBox>
      )
    );
  };

  return (
    statusCode && (
      <Box mt={{ xs: 9, sm: 9, md: 14 }}>
        <Card classes={{ root: classes.cardRoot }}>
          <CardContent classes={{ root: classes.cardContent }}>
            <Grid>
              <GridItem columns={{ xs: 12, sm: 12, md: 3 }}>
                <Box display="flex" justifyContent="center">
                  <img src={backgroundImage} alt="Background" />
                </Box>
              </GridItem>
              <GridItem offset={{ md: 1 }} columns={{ xs: 12, sm: 12, md: 8 }}>
                {renderHeader()}
                {renderRemark()}
                <Details details={claimSubmissionDetails(claimDetails, intl)} />
                <Details
                  header={formatMessage(
                    intl,
                    'claims.claimDetails.header.patientDetails',
                    'Patient details',
                  )}
                  details={patientDetails(
                    membersMap[claimDetails.claimantId],
                    intl,
                  )}
                />
                <Details
                  header={formatMessage(
                    intl,
                    'claims.claimDetails.header.claimDetails',
                    'Claim details',
                  )}
                  details={claimDetailsWithIntl(claimDetails, intl)}
                />
              </GridItem>
              <GridItem offset={{ md: 4 }} columns={{ xs: 12, sm: 12, md: 3 }}>
                <Box mt={{ xs: 4, md: 4 }}>
                  <Button
                    data-testid="btn-back-to-previous-page"
                    variant="contained"
                    color="primary"
                    onClick={router.back}
                  >
                    {formatMessage(
                      intl,
                      'claims.claimDetails.btn.back',
                      'Back',
                    )}
                  </Button>
                </Box>
              </GridItem>
            </Grid>
          </CardContent>
        </Card>
      </Box>
    )
  );
};

ClaimDetails.propTypes = {
  intl: PropTypes.shape({}).isRequired,
  classes: PropTypes.exact({
    cardRoot: PropTypes.string.isRequired,
    cardContent: PropTypes.string.isRequired,
    remark: PropTypes.string.isRequired,
    moreInfo: PropTypes.string.isRequired,
  }).isRequired,
  router: PropTypes.shape({}).isRequired,
  claimDetails: PropTypes.shape({
    status: PropTypes.string.isRequired,
    statusCode: PropTypes.string.isRequired,
    claimSubmissionDate: PropTypes.string.isRequired,
    claimantId: PropTypes.string.isRequired,
    contactNumber: PropTypes.string.isRequired,
    claimItemCategoryCode: PropTypes.string.isRequired,
    consultationType: PropTypes.string.isRequired,
    receiptAmount: PropTypes.number.isRequired,
    reimbursedAmount: PropTypes.number,
    diagnosis: PropTypes.string.isRequired,
    consultationDate: PropTypes.string.isRequired,
    settlementDate: PropTypes.string,
    claimAmountOtherInsurer: PropTypes.number,
    receipts: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
        contentType: PropTypes.string,
      }),
    ).isRequired,
    referrals: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
        contentType: PropTypes.string,
      }),
    ).isRequired,
    isCashlessClaim: PropTypes.bool,
    isMaternity: PropTypes.bool,
  }).isRequired,
  membersMap: PropTypes.shape({}).isRequired,
  getNewClaimDetails: PropTypes.func.isRequired,
  getMemberProfile: PropTypes.func.isRequired,
  claims: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default withStyles(Styles)(injectIntl(withRouter(ClaimDetails)));
