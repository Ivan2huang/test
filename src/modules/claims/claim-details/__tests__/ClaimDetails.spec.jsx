/* eslint-disable no-unused-vars,react/prop-types,react/no-array-index-key */
import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import ClaimDetails from '../ClaimDetails';
import withTheme from '../../../../themes/withThemeProvider';
import withIntl from '../../../../i18n/withIntlProvider';

jest.mock(
  '../../../../uiComponents/Typography',
  // eslint-disable-next-line react/prop-types
  () => ({ children, props }) => (
    <div {...props}>
      Typography Component
      {children}
    </div>
  ),
);

jest.mock('../helper', () => ({
  patientDetails: jest.fn((patientName, contactNumber) => [
    {
      label: 'Patient name',
      value: patientName,
    },
  ]),
  getBackgroundImage: jest.fn(() => '/static/pending_claim_background.svg'),
  claimSubmissionDetails: jest.fn(() => [
    {
      label: 'Claim number',
      value: '1121',
    },
    {
      label: 'Claim submission date',
      value: '16 Aug 2019',
    },
  ]),
  claimDetailsWithIntl: jest.fn(() => [
    {
      label: 'Consultation type',
      value: 'dummy',
    },
    {
      label: 'Consultation date',
      value: '01 Aug 2019',
    },
    {
      label: 'Claim amount',
      value: 'HK$ 123.01',
    },
    {
      label: 'Claim amount(other insurer)',
      value: 'HK$ 123',
    },
  ]),
}));

jest.mock('../Details', () => ({ header, details }) => (
  <div>
    <div>Dummy Details Component</div>
    <div>header</div>
    <div data-id="props">
      {details.map((detail, index) => (
        <div key={index}>
          <span>{detail.label}</span>
          <span>{detail.value}</span>
        </div>
      ))}
    </div>
  </div>
));

jest.mock(
  '../../../../uiComponents/FileItems',
  // eslint-disable-next-line react/prop-types
  () => ({ onClick, ...rest }) => (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions
    <div
      onClick={() => onClick({ name: 'file-name', url: 'file-url' })}
      data-testid="file-items"
      {...rest}
    >
      Dummy FileComponent
    </div>
  ),
);

jest.mock('../../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
  formatAmount: jest.fn((intl, amount) => amount),
  openFile: jest.fn(),
}));

jest.mock('../../../../helpers/mui', () => ({
  useBreakpoint: jest.fn(() => 'xs'),
}));

describe('ClaimDetails component', () => {
  let props;
  const Component = withTheme(withIntl(ClaimDetails));
  beforeEach(() => {
    props = {
      router: {
        query: {
          id: 12,
        },
      },
      getNewClaimDetails: jest.fn(),
      getMemberProfile: jest.fn(),
      claimDetails: {
        status: 'Pending',
        statusCode: 'PROCESSING',
        claimSubmissionDate: '11 Sep 2019',
        claimantId: '3',
        contactNumber: '123456',
        claimItemCategoryCode: 'outpatient',
        consultationType: 'Outpatient',
        diagnosis: 'Abdominal Colic',
        consultationDate: '11 Sep 2019',
        receiptAmount: 1,
        reimbursedAmount: 0,
        settlementDate: '11 Sep 2019',
        claimAmountOtherInsurer: 0,
        remark: 'this is a test remark',
        receipts: [
          {
            name: 'receipt.jpg',
            url: '/receipt.jpg',
          },
        ],
        referrals: [
          {
            name: 'referral.jpg',
            url: '/referral.jpg',
          },
        ],
      },
      membersMap: { 3: 'William Brown' },
      claims: [],
    };
  });
  it('should match the snapshot with Pending claims', () => {
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
  it('should match the snapshot with Approved claims', () => {
    props.claimDetails.statusCode = 'APPROVED';
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
  it('should match the snapshot with Rejected claims', () => {
    props.claimDetails.statusCode = 'REJECTED';
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot with Request for information', () => {
    props.claimDetails.statusCode = 'REQUEST FOR INFORMATION';
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot without before claim data fetched', () => {
    props.claimDetails.statusCode = '';
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot when claims list available', () => {
    props.claims = [
      {
        id: '12',
        status: 'Approved',
        statusCode: 'APPROVED',
        consultationDate: '2019-09-11T04:34:18.065384',
        consultationTypes: 'General practitioner',
        patientId: '3',
        claimedAmount: 1,
        approvedAmount: 0,
        originalClaim: {
          claimId: '12',
          clientId: 'cxadevclient1',
          memberId: '3',
          claimantId: '3',
          receiptDate: '2019-09-11T04:34:18.065384',
          amount: 1.0,
          reason: 'Abdominal Colic',
          reasonCode: 'COLIC',
          type: 'General practitioner',
          typeCode: 'MO-GP',
          category: 'Outpatient',
          categoryCode: 'outpatient',
          status: 'Approved',
          statusCode: 'APPROVED',
          approvedAmount: 0.0,
          acceptTermsAndConditions: true,
          contactNumber: '123456',
          otherInsurerAmount: 0.0,
          documents: {
            referrals: [],
            receipts: [
              {
                id: '1',
                contentType: 'application/pdf',
              },
            ],
          },
        },
      },
    ];
    const { container } = render(<Component {...props} />);

    expect(container).toMatchSnapshot();
  });
});
