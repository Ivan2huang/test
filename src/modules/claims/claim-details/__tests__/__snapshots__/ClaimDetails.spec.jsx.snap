// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimDetails component should match the snapshot when claims list available 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-696"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root injectIntl(withRouter(ClaimDetails))-cardRoot-691 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root injectIntl(withRouter(ClaimDetails))-cardContent-692"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-815"
            >
              <img
                alt="Background"
                src="/static/pending_claim_background.svg"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-816"
            >
              <div>
                Typography Component
                Processing
              </div>
              <div>
                Typography Component
                Wellness Claim
              </div>
              <div>
                Typography Component
                this is a test remark
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Claim number
                  </span>
                  <span>
                    1121
                  </span>
                </div>
                <div>
                  <span>
                    Claim submission date
                  </span>
                  <span>
                    16 Aug 2019
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Patient name
                  </span>
                  <span>
                    William Brown
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Consultation type
                  </span>
                  <span>
                    dummy
                  </span>
                </div>
                <div>
                  <span>
                    Consultation date
                  </span>
                  <span>
                    01 Aug 2019
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount
                  </span>
                  <span>
                    HK$ 123.01
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount(other insurer)
                  </span>
                  <span>
                    HK$ 123
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-817"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                data-testid="btn-back-to-previous-page"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Back
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClaimDetails component should match the snapshot with Approved claims 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-174"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root injectIntl(withRouter(ClaimDetails))-cardRoot-169 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root injectIntl(withRouter(ClaimDetails))-cardContent-170"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-293"
            >
              <img
                alt="Background"
                src="/static/pending_claim_background.svg"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-294"
            >
              <div>
                Typography Component
                Approved
              </div>
              <div>
                Typography Component
                Wellness Claim
              </div>
              <div>
                Typography Component
                this is a test remark
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Claim number
                  </span>
                  <span>
                    1121
                  </span>
                </div>
                <div>
                  <span>
                    Claim submission date
                  </span>
                  <span>
                    16 Aug 2019
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Patient name
                  </span>
                  <span>
                    William Brown
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Consultation type
                  </span>
                  <span>
                    dummy
                  </span>
                </div>
                <div>
                  <span>
                    Consultation date
                  </span>
                  <span>
                    01 Aug 2019
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount
                  </span>
                  <span>
                    HK$ 123.01
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount(other insurer)
                  </span>
                  <span>
                    HK$ 123
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-295"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                data-testid="btn-back-to-previous-page"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Back
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClaimDetails component should match the snapshot with Pending claims 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root injectIntl(withRouter(ClaimDetails))-cardRoot-1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root injectIntl(withRouter(ClaimDetails))-cardContent-2"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-125"
            >
              <img
                alt="Background"
                src="/static/pending_claim_background.svg"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-126"
            >
              <div>
                Typography Component
                Processing
              </div>
              <div>
                Typography Component
                Wellness Claim
              </div>
              <div>
                Typography Component
                this is a test remark
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Claim number
                  </span>
                  <span>
                    1121
                  </span>
                </div>
                <div>
                  <span>
                    Claim submission date
                  </span>
                  <span>
                    16 Aug 2019
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Patient name
                  </span>
                  <span>
                    William Brown
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Consultation type
                  </span>
                  <span>
                    dummy
                  </span>
                </div>
                <div>
                  <span>
                    Consultation date
                  </span>
                  <span>
                    01 Aug 2019
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount
                  </span>
                  <span>
                    HK$ 123.01
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount(other insurer)
                  </span>
                  <span>
                    HK$ 123
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-127"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                data-testid="btn-back-to-previous-page"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Back
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClaimDetails component should match the snapshot with Rejected claims 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-342"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root injectIntl(withRouter(ClaimDetails))-cardRoot-337 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root injectIntl(withRouter(ClaimDetails))-cardContent-338"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-461"
            >
              <img
                alt="Background"
                src="/static/pending_claim_background.svg"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-462"
            >
              <div>
                Typography Component
                Rejected
              </div>
              <div>
                Typography Component
                Wellness Claim
              </div>
              <div>
                Typography Component
                this is a test remark
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Claim number
                  </span>
                  <span>
                    1121
                  </span>
                </div>
                <div>
                  <span>
                    Claim submission date
                  </span>
                  <span>
                    16 Aug 2019
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Patient name
                  </span>
                  <span>
                    William Brown
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Consultation type
                  </span>
                  <span>
                    dummy
                  </span>
                </div>
                <div>
                  <span>
                    Consultation date
                  </span>
                  <span>
                    01 Aug 2019
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount
                  </span>
                  <span>
                    HK$ 123.01
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount(other insurer)
                  </span>
                  <span>
                    HK$ 123
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-463"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                data-testid="btn-back-to-previous-page"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Back
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClaimDetails component should match the snapshot with Request for information 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-510"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root injectIntl(withRouter(ClaimDetails))-cardRoot-505 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root injectIntl(withRouter(ClaimDetails))-cardContent-506"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-629"
            >
              <img
                alt="Background"
                src="/static/pending_claim_background.svg"
              />
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-630"
            >
              <div>
                Typography Component
                Processing
              </div>
              <div>
                Typography Component
                Wellness Claim
              </div>
              <div>
                Typography Component
                this is a test remark
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-634 makeStyles-container-632 makeStyles-container-633 NotificationBox-container-631 injectIntl(withRouter(ClaimDetails))-moreInfo-508"
            >
              <div
                class="MuiBox-root MuiBox-root-635"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  size="24"
                  style="width: 1.125rem; height: 1.125rem;"
                  viewBox="0 0 18 18"
                >
                  <path
                    d="M8.30911 1.68243L0.109113 15.8004C-0.199887 16.3324 0.184113 17.0004 0.800113 17.0004H17.2001C17.8161 17.0004 18.2001 16.3324 17.8911 15.8004L9.69111 1.68243C9.38311 1.15143 8.61711 1.15143 8.30911 1.68243Z"
                    fill="#A8000B"
                  />
                  <path
                    d="M9.89997 5H8.09998V11H9.89997V5Z"
                    fill="white"
                  />
                  <path
                    d="M9 14.826C9.69036 14.826 10.25 14.2663 10.25 13.576C10.25 12.8856 9.69036 12.326 9 12.326C8.30964 12.326 7.75 12.8856 7.75 13.576C7.75 14.2663 8.30964 14.826 9 14.826Z"
                    fill="white"
                  />
                </svg>
              </div>
              <div>
                Typography Component
                More information is required. We will contact you soon.
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Claim number
                  </span>
                  <span>
                    1121
                  </span>
                </div>
                <div>
                  <span>
                    Claim submission date
                  </span>
                  <span>
                    16 Aug 2019
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Patient name
                  </span>
                  <span>
                    William Brown
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div>
                Dummy Details Component
              </div>
              <div>
                header
              </div>
              <div
                data-id="props"
              >
                <div>
                  <span>
                    Consultation type
                  </span>
                  <span>
                    dummy
                  </span>
                </div>
                <div>
                  <span>
                    Consultation date
                  </span>
                  <span>
                    01 Aug 2019
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount
                  </span>
                  <span>
                    HK$ 123.01
                  </span>
                </div>
                <div>
                  <span>
                    Claim amount(other insurer)
                  </span>
                  <span>
                    HK$ 123
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
          >
            <div
              class="MuiBox-root MuiBox-root-645"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                data-testid="btn-back-to-previous-page"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Back
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClaimDetails component should match the snapshot without before claim data fetched 1`] = `
<div>
  
</div>
`;
